test/unit/tarantool_role_test.lua

```diff
@@ -0,0 +1,555 @@
+        }, {is_master = false})
+
+        fiber.sleep(0.5)
+        t.assert_not_equals(expirationd.task(task_name1), nil)
+        t.assert_equals(expirationd.cfg.metrics, value)
+    end
+end
+
+function g.test_apply_config_start_cfg_task(cg)
+    local task_name = "cfg"
+    cg.role.apply({
+            [task_name] = {
+                space = g.space.id,
+                is_expired = always_true_func_name,
+            },
+    }, {is_master = false})
+
+    fiber.sleep(0.5)
+    t.assert_not_equals(expirationd.task(task_name), nil)
+end
+
+function g.test_apply_config_start_cfg_task_with_box_func(cg)
+    local task_name = 'cfg'
+    cg.role.apply({
+            [task_name] = {
+                space = g.space.id,
+                is_expired = always_true_func_name_in_box_func
+            }
+    }, { is_master = false })
+
+    fiber.sleep(0.5)
+    t.assert_not_equals(expirationd.task(task_name), nil)
+end
+
+function g.test_apply_config_start_cfg_task_with_correct_order(cg)
+    t.assert_not(rawget(_G, always_true_func_name_with_side_effect_flag))
+    local task_name = 'cfg'
+    cg.space:insert({1, "1"})
+    cg.role.apply({
+        [task_name] = {
+            space = g.space.id,
+            is_expired = always_true_func_name_with_side_effect
+        }
+    }, { is_master = false })
+
+    fiber.sleep(0.5)
+    t.assert_not_equals(expirationd.task(task_name), nil)
+    t.assert(rawget(_G, always_true_func_name_with_side_effect_flag))
+end
+
+function g.test_apply_config_start_task_with_all_options(cg)
+    local task_name = "apply_config_test"
+    local options =  {
+        args = {"any"},
+        atomic_iteration = false,
+        force = false,
+        force_allow_functional_index = true,
+        full_scan_delay = 1,
+        full_scan_time = 1,
+        index = 0,
+        iterate_with = iterate_pairs_func_name,
+        iteration_delay = 1,
+        iterator_type = "ALL",
+        on_full_scan_complete = always_true_func_name,
+        on_full_scan_error = always_true_func_name,
+        on_full_scan_start = always_true_func_name,
+        on_full_scan_success = always_true_func_name,
+        process_expired_tuple = always_true_func_name,
+        process_while = always_true_func_name,
+        start_key = {1},
+        tuples_per_iteration = 100,
+        vinyl_assumed_space_len_factor = 1,
+        vinyl_assumed_space_len = 100,
+    }
+
+    cg.role.apply({
+        [task_name] = {
+            space = g.space.id,
+            is_expired = always_true_func_name,
+            options = options,
+        }
+    }, {is_master = true})
+
+    fiber.sleep(0.5)
+    t.assert_not_equals(expirationd.task(task_name), nil)
+
+    -- pass an invalid option
+    options.full_scan_time = -100
+    pcall(cg.role.apply, {
+        [task_name] = {
+            space = g.space.id,
+            is_expired = always_true_func_name,
+            options = options,
+        }
+    }, {is_master = true})
+    fiber.sleep(0.5)
```

Comment:
Please, replace the sleep with `helpers.retrying`. It will make test less flaky.